### PR TITLE
Add unittests and fix bugs

### PR DESCRIPTION
**Pull request by jwhonce**
* Improved error messages
* Improved checking of user input

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
This is the promised clean up, bug fixes and more unit tests

**Comment by mheon** [comment]
@jwhonce Would you mind changing the commit messages to prefix them with "Python:" or similar? We're generating changelogs out of our commit logs, so it would be nice if we could distinguish where commits are going.

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, going to kick the bot as the tests other than travis failed without leaving any details.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please.

**Comment by umohnani8** [comment]
LGTM

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM
But need to get tests working.

**Comment by TomSweeneyRedHat** [comment]
My PR just failed these same three tests with the same "internal error".  Is there someone we can talk to about them?  Looks like some kind of test environment issue going around.

**Comment by mheon** [comment]
@jlebon Is papr down? We're seeing consistent "Internal Error" failures, which I'm assuming means provisioning VMs?

**Comment by jlebon** [comment]
Yes, it's down. IT is investigating OpenStack failures.

File: contrib/python/podman/libs/_containers_attach.py [hunk 1]
```diff
@@ -73,38 +76,37 @@ def log_handler(signum, frame):
             # catch any resizing events and send the resize info
             # to the control fifo "socket"
             signal.signal(signal.SIGWINCH, resize_handler)
+
         except termios.error:
             original_attr = None
 
         try:
-            # Prepare socket for communicating with conmon/container
+            # TODO: socket.SOCK_SEQPACKET may not be supported in Windows
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply contrib/python/podman/libs/_containers_attach.py]: @TomSweeneyRedHat Cool!  I'll leave this warning to check on this when we port.

**Comment by jwhonce** [comment]
@mheon Will do.  How about using the package name as a label  "[python-podman]"  ?  /cc @baude

**Comment by mheon** [comment]
@jwhonce SGTM

**Comment by mheon** [comment]
bot, retest this please

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
LGTM

**Pull request closed without merging**